apps/voting/contracts/Voting.sol

```diff
@@ -361,14 +425,39 @@ contract Voting is IForwarder, AragonApp {
 
     function _canVote(uint256 _voteId, address _voter) internal view returns (bool) {
         Vote storage vote_ = votes[_voteId];
-
         return _isVoteOpen(vote_) && token.balanceOfAt(_voter, vote_.snapshotBlock) > 0;
     }
 
+    function _canVoteOnBehalfOf(uint256 _voteId, address _principal, address _representative) internal view returns (bool) {
```

Comment:
Wouldn't it make sense to have `_voteExists` as acondition inside this funcion too?

Reply:
I don't think it is necessary since all the consumers of this function have already checked it, but we should make a comment about this

Reply:
But the function is internal, not private, so it could be misleading/dangerous. 
If I'm not wrong, the way it is right now it could happen that this function is called for a vote that doesn't exists and returns true, which would be formally correct from a logical point of view but wouldn't make any semantic sense in my opinion. (And if I'm wrong, then we should remove the `voteExists` check from the public wrapper).
Besides, generally speaking I like better having these kind of checks here in the inner part, although we already talked about that.

Reply:
There are more than one internal function receiving a vote id without checking the vote actually exists. However, I see your point but I also think we should find a way of performing that check only once. In that case, we can turn the internal functions to receive a `Vote` per se instead of an id, assuming the consumer already checked the given storage pointer is valid.